File: BackEnd/news_service/src/main/java/Myproject/news_service/entity/News.java
package Myproject.news_service.entity;


import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class News {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int newsId;

    private String newsName; // tên
    private String newsTitle; //tiêu đề
    private String newsDate; // thời gian  ngày
    private String newsTime; //  giờ
    private String newsCategory; // loại tin
    private String newsImage;
    private String newsImage1;
    private String newsImage2;
    private String newsImage3;
    private String newsContent;
    private String newsContent1;
    private String newsContent2;
    private String newsContent3;
}

File: FrontEnd/javaScript/News.js
const params = new URLSearchParams(window.location.search);
const newsId = params.get('id');


fetch(`http://localhost:8083/NewsDatabase/news/${newsId}`)
    .then(res => {
        if (!res.ok) throw new Error("Không tìm thấy tin tức");
        return res.json();
    })
    .then(news => {
        const newsDiv = document.getElementById("section_two_content");
        newsDiv.innerHTML = " ";
        newsDiv.innerHTML = `
            <div class="news-detail">
                <div class="news-detail-content">
                    <h2>${news.newsTitle}</h2>
                    <div><b>Ngày đăng:</b> ${news.newsDate}</div>
                    <a href="javascript:void(0)" onclick="window.location.href='../html/Product.html?id=3'">
                        <img src="${news.newsImage}" alt="${news.newsTitle}" class="news-detail-image">
                    </a>
                    <p>${news.newsContent}</p>
                </div>

                <div class="news-detail-content">
                    <img src="${news.newsImage1}" alt="${news.newsTitle1}" class="news-detail-image">
                    <p>${news.newsContent1}</p>
                </div>

                <div class="news-detail-content">
                    <img src="${news.newsImage2}" alt="${news.newsTitle2}" class="news-detail-image">
                    <p>${news.newsContent2}</p>
                </div>
                <div class="news-detail-content">
                    <img src="${news.newsImage3}" alt="${news.newsTitle3}" class="news-detail-image">
                    <p>${news.newsContent3}</p>
                </div>

            </div>
        `;
    })
    .catch(error => console.error("Lỗi khi load dữ liệu từ API:", error));